src/mongo/Config.class.php

```diff
@@ -1759,12 +1760,47 @@ protected function getConnectionForDataSource($dataSource)
         }
         if(!isset($this->connections[$dataSource]))
         {
-            $this->connections[$dataSource] = new \MongoClient($ds['connection'], $connectionOptions);
+            $retries = 1;
+            $exception = null;
+            $connected = false;
+
+            do {
+                try {
+                    $this->connections[$dataSource] = $this->getMongoClient($ds['connection'], $connectionOptions);
+                    // Check if we are connected
+                    $connections = $this->connections[$dataSource]->getConnections();
+                    if (empty($connections) === false) {
+                        $connected = true;
+                    }
+                } catch (\MongoConnectionException $e) {
+                    self::getLogger()->warn("MongoConnectionException: " . $e->getMessage());
```

Comment:
I'd like to see a little more information in the log.

Maybe something like:
```
"Connection attempt 1 failed. Retrying..."
"Connection attempt 2 failed. Retrying..."
"Connection attempt 30 failed. Max Retry reached."
```

This would give us more information on how often this was happening, how well we recovered.